src/super_gradients/training/models/segmentation_models/ppliteseg.py

```diff
@@ -0,0 +1,383 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from typing import Union, List, Tuple
+from super_gradients.training.utils.module_utils import ConvBNReLU, UpsampleMode, make_upsample_module
+from super_gradients.training.models.segmentation_models.stdc import SegmentationHead, AbstractSTDCBackbone, STDC1Backbone, STDC2Backbone
+from super_gradients.training.models.segmentation_models.segmentation_module import SegmentationModule
+from super_gradients.training.utils import HpmStruct, get_param
+
+
+class SPPM(nn.Module):
+    """
+    Simple Pyramid Pooling context Module.
+    """
+    def __init__(self,
+                 in_channels: int,
+                 inter_channels: int,
+                 out_channels: int,
+                 pool_sizes: List[Union[int, Tuple[int, int]]],
+                 upsample_mode: Union[UpsampleMode, str] = UpsampleMode.BILINEAR,
+                 align_corners: bool = False):
+        """
+        :param inter_channels: num channels in each pooling branch.
+        :param out_channels: The number of output channels after pyramid pooling module.
+        :param pool_sizes: spatial output sizes of the pooled feature maps.
+        """
+        super().__init__()
+        self.branches = nn.ModuleList([
+            nn.Sequential(
+                nn.AdaptiveAvgPool2d(pool_size),
+                ConvBNReLU(in_channels, inter_channels, kernel_size=1, bias=False),
+            ) for pool_size in pool_sizes
+        ])
+        self.conv_out = ConvBNReLU(inter_channels, out_channels, kernel_size=3, padding=1, bias=False)
+        self.out_channels = out_channels
+        self.upsample_mode = upsample_mode
+        self.align_corners = align_corners
+        self.pool_sizes = pool_sizes
+
+    def output_channels(self):
+        return self.out_channels
```

Comment:
Since SPPM is hard-coded I don't really need this specific method. I was probably thinking of a future structure and factory for context modules.
But it is a work for a different PR.
Remove `out_channels()` method, simply call `sppm.out_channels` [bd37f20](https://github.com/Deci-AI/super-gradients/pull/349/commits/bd37f20eb277d8bd5ae6a0b33b1fbbd5a3315c08)